rai/node/node.cpp

```diff
@@ -3105,6 +3106,37 @@ bool rai::handle_node_options (boost::program_options::variables_map & vm)
 			std::cerr << "Vacuum failed" << std::endl;
 		}
 	}
+	else if (vm.count ("snapshot"))
+	{
+		try
+		{
+			boost::filesystem::path data_path = vm.count ("data_path") ? boost::filesystem::path (vm["data_path"].as<std::string> ()) : rai::working_path ();
+
+			auto source_path = data_path / "data.ldb";
+			auto snapshot_path = data_path / "snapshot.ldb";
+
+			std::cout << "Database snapshot of " << source_path << " to " << snapshot_path << " in progress" << std::endl;
+			std::cout << "This may take a while..." << std::endl;
+
+			bool success = false;
+			{
+				inactive_node node (data_path);
+				success = node.node->copy_with_compaction (snapshot_path);
+			}
+			if (success)
+			{
+				std::cout << "Snapshot completed, This can be found at " << snapshot_path << std::endl;
```

Comment:
If this fails for some reason other than an exception, we won't print anything to stderr and will probably give an exitcode of zero to the parent.